### PR TITLE
AdminClientView page

### PR DESCRIPTION
**Pull request by Puepis**
[Notion task](https://www.notion.so/uwblueprintexecs/34a44be943164278bfbf43de35e4ef7b?v=448f1b24885146a7b264dbf09eb60be1&p=e83a70b382bd40e5993d8cfa7a48092e)
[Figma breakdown](https://www.figma.com/file/7f5qYOgY3oaSHC5OMtuscw/S21-Dev-Onboarding-Task%3A-Client-Table?node-id=0%3A1)

Backend changes
- added query and implemented resolver to grab requests by client first name

Frontend changes
- built the AdminClientView page to display the client's requests using the `ClientRequestTable` and added css styling
- fetched client requests using the new backend query

### Normal view
normal

### All requests fulfilled view
all-requests-checked

### No requests view
no-existing-requests

### CODE REVIEW

File: client/src/pages/style/AdminClientView.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.admin-client-view-page-wrapper {
+    display: flex;
+    flex-direction: column;
+    padding-left: 160px;
+
+    .admin-client-view-header {
```
Comment (guy-ro) [inline client/src/pages/style/AdminClientView.scss:6]: Simply

.admin-client-view-client-name {
    @include heavy-text;
}

File: client/src/pages/style/AdminClientView.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            .admin-client-view-client-name {
+                @include heavy-text;
+            }
+        }
+    }
+}
+
+
+.admin-client-view-page-content {
+    padding-right: 154px;
```
Comment (guy-ro) [inline client/src/pages/style/AdminClientView.scss:17]: Is this to center align? If so, align it in center

File: client/src/pages/AdminClientView.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  const [firstName, lastName] = fullName.split("-");
+  const [requests, setRequests] = useState<Request[]>([]);
+  const [numUnfulfilledRequests, setNumUnfulfilledRequests] = useState(0);
+  const [noExistingRequests, setNoExistingRequests] = useState(false);
+
+  const query = gql`
+    query getClientRequests($firstName: String) {
+      getClientRequests(firstName: $firstName) {
+        _id
+        requestId
```
Comment (guy-ro) [inline client/src/pages/AdminClientView.tsx:25]: Don't fetch things you don't use

**Review by guy-ro — APPROVED** [review]
A few comments, but looks good!

File: server/graphql/resolvers.ts [hunk 4]
```diff
@@ -28,6 +28,10 @@ const resolvers = {
     }),
     clients: (_, __, { dataSources, authenticateUser }): Array<ClientInterface> => authenticateUser().then(() => dataSources.clients.getAll()),
     request: (_, { id }, { dataSources }): RequestInterface => dataSources.requests.getById(Types.ObjectId(id)),
+    getClientRequests: (_, {firstName}, {dataSources}): Array<RequestInterface> => dataSources.requests.getAll().filter(request => {
```
Comment (meganniu) [inline server/graphql/resolvers.ts:31]: Only allow authorized users to access client info. See [this line](https://github.com/uwblueprint/pregnancy-centre/blob/4f49f5d367b2a270080c2e93a07981e993b1e2c5/server/graphql/resolvers.ts#L29) for how

Comment (meganniu) [inline server/graphql/resolvers.ts:31]: I'm curious about why we're only getting clients by first name

**Review by meganniu — COMMENTED** [review]
Amazing work @Puepis!!! Can't wait to see what u accomplish this term :')

left a few comments bc guy has already addressed the big concerns

File: server/graphql/resolvers.ts (hunk 4 above)
Comment (Puepis) [reply server/graphql/resolvers.ts:31]: oh in the task description it said to fetch all of a client's requests given their first name so I just followed it

**Pull request closed without merging**